src/main/java/io/vertx/core/http/impl/ClientConnection.java

```diff
@@ -183,46 +187,48 @@ public void channelInactive(ChannelHandlerContext ctx) throws Exception {
 
     @Override
     public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
-      context.executeFromIO(() -> {
-        synchronized (ClientConnection.this) {
-          if (handshaker != null && handshaking) {
-            if (msg instanceof HttpResponse) {
-              HttpResponse resp = (HttpResponse) msg;
-              if (resp.getStatus().code() != 101) {
+      synchronized (ClientConnection.this) {
```

Comment:
At this point the thread is an event loop thread. So it means we need it in the executeFromIO smaller blocks.